eng/common/TestResources/New-TestResources.ps1

```diff
@@ -385,7 +385,15 @@ foreach ($templateFile in $templateFiles) {
 
     Log "Deploying template '$templateFile' to resource group '$($resourceGroup.ResourceGroupName)'"
     $deployment = Retry {
-        New-AzResourceGroupDeployment -Name $BaseName -ResourceGroupName $resourceGroup.ResourceGroupName -TemplateFile $templateFile -TemplateParameterObject $templateFileParameters
+        $lastDebugPreference = $DebugPreference
+        try {
+            $DebugPreference = "Continue"
```

Comment:
Don't do this by default. If users want, they can pass `-Debug` to the script that should carry through automatically. That's idiomatic and not intrusive to everyone. Debug spew can be way too much.

Reply:
```powershell
function foo { [CmdletBinding()] param() function bar { [CmdletBinding()] param() Write-Debug "Debug message" } bar }
foo # nothing returned
foo -debug # DEBUG: Debug message
```

Reply:
If you wanted to do this in CIs (either in the YAML template or if `$CI`), sure: dumping tons of information for CIs can be useful rather than rerunning). But it's not a good customer UX normally.

Reply:
The problem I faced is I don't want Debug enabled for all commands, just this one. Passing `-Debug` through for CI scenarios does not accomplish this. I can do a local conditional right here to set debug if `$CI -eq $true`?